### ISSUE TITLE
Count does not work on an empty grouped record set

### ISSUE THREAD

**Issue by shopblocks** [UnionOfRAD/lithium#1281]
Performing a `Product::count(...)` throws an error:

```
[message:protected] => Undefined offset: 0
    [string:Exception:private] => 
    [code:protected] => 8
    [file:protected] => /var/www/vhosts/miribase/libraries/lithium/data/source/Database.php
    [line:protected] => 745
    [trace:Exception:private] => Array
```

Example query:
```sql
SELECT COUNT(*) as count FROM `products` AS `Products` 
LEFT JOIN `product_categories` AS `ProductCategories` ON `Products`.`product_category_id` = `ProductCategories`.`id` 
LEFT JOIN `brands` AS `Brands` ON `Products`.`brand_id` = `Brands`.`id` 
LEFT JOIN `tags_models` AS `TagsModels` ON `TagsModels`.`model` = "products" AND `Products`.`id` = `TagsModels`.`model_id` 
LEFT JOIN `tags` AS `Tags` ON `TagsModels`.`tag_id` = `Tags`.`id` 
LEFT JOIN `products` AS `Products__2` ON `Products`.`id` = `Products__2`.`parent_id` 
WHERE 
`Products`.`deleted` = 0 
AND `Products`.`extra` = 0 
AND `Products`.`preview` = 0 
AND Products.parent_id IS NULL 
AND `Products__2`.`deleted` = 0 
GROUP BY `Products__2`.`parent_id` 
ORDER BY `Products`.`created` DESC;
```

Returns no rows.

![image](https://cloud.githubusercontent.com/assets/14819903/16124062/5559cc60-33e5-11e6-8513-2491a916caf4.png)

**Comment by shopblocks** [comment]
Error only happens if `group` is used.

Whether or not it is relevant that we are grouping on ourself.

**Comment by davidpersson** [comment]
This might be a legitimate issue. The query seems to be quite complex and makes it hard to debug. 

Please provide a minimal case that still shows the issue: 1. Remove all the relationships. 2. Remove all ordering an 3. minimalize the conditions. Thanks!

**Comment by shopblocks** [comment]
Some basic code to reproduce.

Get any model, with a table with no results. This is being tested against MySQL 5.6, no idea if the same issue exists in other drivers.

```php
ContentBlocks::count([
            'conditions' => [
                'deleted' => 0,
            ],
            'group' => 'name',
        ]);
```

Structure is as follows, though I don't think the structure matters.

![image](https://cloud.githubusercontent.com/assets/14819903/16146942/89136514-3478-11e6-872d-b2c428e72fbb.png)


In this I also found another issue, which i'll put in a separate thread.

**Comment by shopblocks** [comment]
#1282

**Comment by davidpersson** [comment]
`calculation()` will now handle empty results correctly -- fixed in 972cd4969e63a627e551e869de1477995311ca43

Please note that your query is returning an empty result. For support on how to structure the query correctly, so that it returns result use stackoverflow.

**Comment by shopblocks** [comment]
We expected the result to have no rows, the bug was only that the notice was being thrown.

Thanks for fixing.

**Comment by shopblocks** [comment]
@davidpersson Any idea when this will be made into a release?

**Comment by davidpersson** [comment]
@shopblocks I guess we'll package the next 1.0.x release end of the month, 1.1.x will be released according to schedule http://li3.me/versions -- fixes from 1.0 are regularly merged into 1.1 if you are using that
